docs/en/cue/development-guide.md

```diff
@@ -0,0 +1,914 @@
+
+For the newcomers, the first parts are some basic CUE knowledge.
+The rest are advanced CUE development tips that will be used in creating KubeVela Definition files.
+
+## Install Tools
+
+We will rely on these following tools:
+
+* [`cue` >=v0.2.2](https://cuelang.org/docs/install/)
+* [`vela` (>v1.0.0)](https://kubevela.io/#/en/install?id=_3-optional-get-kubevela-cli)
+
+Please install them first before continue.
+
+## CUE CLI basic
+
+Below is the basic CUE data, you can define both schema and value in the same file with the almost same format:
+
+```
+a: 1.5
+a: float
+b: 1
+b: int
+d: [1, 2, 3]
+g: {
+	h: "abc"
+}
+e: string
+```
+
+Let's write them in a file named `first.cue`.
+
+* Format the CUE file. If you're using Goland or similar JetBrains IDE,
+  you can [configure save on format](https://wonderflow.info/posts/2020-11-02-goland-cuelang-format/) instead.
+  This command will not only format the CUE, but also point out the wrong schema. That's very useful.
+    ```shell
+    cue fmt first.cue
+    ```
+
+* Schema Check, besides `cue fmt`, you can also use `vue vet` to check schema.
+    ```shell
+    cue vet first.cue
+    ```
+  
+* Calculate/Render the result. `cue eval` will calculate the CUE file and render out the result.
+  You can see the results don't contain `a: float` and `b: int`, because these two variables are calculated.
+  While the `e: string` doesn't have definitive results, so it keeps as it is.
+    ```shell
+   $ cue eval first.cue
+    a: 1.5
+    b: 1
+    d: [1, 2, 3]
+    g: {
+    h: "abc"
+    }
+    e: string
+    ```
+  
+* Render for specified result. For example, we want only know the result of `b` in the file, then we can specify the parameter `-e`.
+    ```shell
+    $ cue eval -e b first.cue
+    1
+    ```
+
+* Export the result. `cue export` will export the result with final value. It will report an error if some variables are not definitive. 
+    ```shell
+    $ cue export first.cue
+    e: cannot convert incomplete value "string" to JSON:
+        ./first.cue:9:4
+    ```
+  We can complete the value by giving a value to `e`, for example:
+    ```shell
+    echo "e: \"abc\"" >> first.cue
+    ```
+  Then, the command will work. By default, the result will be rendered in json format. 
+    ```shell
+    $ cue export first.cue
+    {
+        "a": 1.5,
+        "b": 1,
+        "d": [
+            1,
+            2,
+            3
+        ],
+        "g": {
+            "h": "abc"
+        },
+        "e": "abc"
+    }
+    ```
+
+* Export the result in YAML format.
+    ```shell
+    $ cue export first.cue --out yaml
+    a: 1.5
+    b: 1
+    d:
+    - 1
+    - 2
+    - 3
+    g:
+      h: abc
+    e: abc
+    ```
+
+* Export the result for specified variable.
+    ```shell
+    $ cue export -e g first.cue
+    {
+        "h": "abc"
+    }
+    ```
+
+For now, you have learned all useful CUE cli operations.
+
+## CUE language basic
```

Comment:
We are either splitting CUE basic to two places or there's duplication: https://github.com/oam-dev/kubevela/blob/e17b3a43841829f14488aa586a24c7fd5d5b5ff8/docs/en/cue/basic.md

Reply:
let's put them togathor in another PR